src/System.Private.CoreLib/shared/System/Resources/ManifestBasedResourceGroveler.cs

```diff
@@ -463,6 +463,27 @@ private void HandleSatelliteMissing()
             throw new MissingSatelliteAssemblyException(SR.Format(SR.MissingSatelliteAssembly_Culture_Name, _mediator.NeutralResourcesCulture, satAssemName), missingCultureName);
         }
 
+        private static string GetManifestResourceNamesList(Assembly assembly)
+        {
+            try
+            {
+                string postfix = "\"";
+                string []? resourceSetNames = assembly.GetManifestResourceNames();
+
+                // If we have more than 10 resource sets, we just print the first 10 for the sake of the exception message readability.
+                if (resourceSetNames.Length > 10)
+                {
+                    Array.Resize(ref resourceSetNames, 10);
```

Comment:
Can this use the new range syntax? Something like: `resourceSetNames = resourceSetNames[..10];`

Reply:
That is a good idea!